lib/Server/ResponseTypes/IdTokenResponse.php

```diff
@@ -97,7 +97,31 @@ protected function getExtraParams(AccessTokenEntityInterface $accessToken)
         $claims = $this->claimExtractor->extract($accessToken->getScopes(), $userEntity->getClaims());
 
         foreach ($claims as $claimName => $claimValue) {
-            $builder->withClaim($claimName, $claimValue);
+            switch ($claimName) {
+                case RegisteredClaims::AUDIENCE:
+                    $builder->permittedFor($claimValue);
+                    break;
+                case RegisteredClaims::EXPIRATION_TIME:
+                    $builder->expiresAt($claimValue);
+                    break;
+                case RegisteredClaims::ID:
+                    $builder->identifiedBy($claimValue);
+                    break;
+                case RegisteredClaims::ISSUED_AT:
+                    $builder->issuedAt(new DateTimeImmutable($claimValue));
```

Comment:
I believe  $claimValue is a unix timestamp. To pass it into `DateTimeImmutable` I believe it needs to be the string `'@' . $claimValue`.  See "Unix Timestamp" under https://www.php.net/manual/en/datetime.formats.compound.php